test_tools/joiner/text_converter.cc

```diff
@@ -0,0 +1,147 @@
+#include <string>
+#include <vector>
+#include <fstream>
+#include <flatbuffers/flatbuffers.h>
+#include <iostream>
+#include "../../rlclientlib/logger/preamble.h"
+#include "../../rlclientlib/logger/message_type.h"
+#include "../../rlclientlib/generated/RankingEvent_generated.h"
+#include "../../rlclientlib/generated/OutcomeEvent_generated.h"
+
+// namespace aliases
+namespace rlog = reinforcement_learning::logger;
+namespace flat = reinforcement_learning::messages::flatbuff;
+////
+
+namespace reinforcement_learning { namespace joiner {
+
+  // forward declarations 
+  void convert_to_text(const std::string& file);
+  void print_ranking_event(void* buff);
+  void print_outcome_event(void* buff);
+  void print_numeric_outcome(const flat::OutcomeEventHolder* evt);
+  void print_string_outcome(const flat::OutcomeEventHolder* evt);
+  void print_action_outcome(const flat::OutcomeEventHolder* evt);
+  ////
+
+  void convert_to_text(const std::vector<std::string>& files) {
+    for(const auto& file:files) {
+      convert_to_text(file);
+    }
+  }
+
+  void convert_to_text(const std::string& file){
+    std::ifstream infile;
+    infile.open(file, std::ios_base::binary);
+    if (infile.fail() || infile.bad()){
+      std::cout << "Unable to open file: " << file << std::endl;
+    }
+    std::cout << "File:" << file << std::endl;
+    do {
+      char raw_preamble[8];
+      if (infile.fail() || infile.bad())
+        return;
```

Comment:
throw?

Reply:
Don't want to stop processing.  Other files might be fine.  Writing to cerr makes sense.